x/cli/introspection.go

```diff
@@ -0,0 +1,169 @@
+package cli
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/upfluence/cfg/internal/help"
+	"github.com/upfluence/cfg/internal/synopsis"
+)
+
+type CommandDefinition struct {
+	Args    []string
+	Configs []interface{}
+}
+
+type IntrospectionOptions struct {
+	AppName     string
+	Definitions []CommandDefinition
+	Short       bool
+
+	args map[string]string
+}
+
+func (io IntrospectionOptions) argName(arg string) string {
```

Comment:
nit: `io` is confusing here imo when you also have `import "io"`.

```suggestion
func (iopts IntrospectionOptions) argName(arg string) string {
```